speedy-recovery-client/src/service/BackendService.js

```diff
@@ -0,0 +1,108 @@
+import {
+  getAlt,
+  getAvatar,
+  getDate,
+  getId,
+  getPosition,
+  getSubtitle,
+  getText,
+  getTime,
+  getTitle,
+  getUnread,
+  getUserId
+} from "./BackendMappingExtractionUtils";
+import axios from "axios";
+
+const baseUrl = "https://speedy-recovery-server.azurewebsites.net";
+
+const getConversation = async id => {
+  const url = `${baseUrl}/conversations?userid=${id}`;
+  try {
+    const response = await axios.get(url);
+    return response.data;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const getMessages = async (id, id2) => {
+  const url = `${baseUrl}/conversation?userid1=${id}&userid2=${id2}`;
+  try {
+    const response = await axios.get(url);
+    return response.data;
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+const getPractitionerInfo = async id => {
+  const url = `${baseUrl}/practitioners?userid=${id}`;
+  try {
+    return await axios.get(url);
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const postMessages = async (senderID, recipientID, message) => {
+  const url = `${baseUrl}/messages?`;
+  const headers = {
+    Sender: senderID,
+    Recipient: recipientID
+  };
+  const data = {
+    message: `<message_start>${message}<message_end>`
+  };
+
+  axios
+    .post(url, data, { headers: headers })
+    .then(response => console.log("POST server success: ", response))
+    .catch(error => console.error("POST server error: ", error));
+};
+
+const mapConversations = (conversationResource, id, conversationList) => ({
```

Comment:
These mapping functions should be extracted into their own adaptors. This would allow support for other data of a different structure to be plugged in

Reply:
@joshuazeltser but these are data from the database not Fhir. so we dont need to work with different adapters?

Reply:
I think for any external data we should use adaptors. For example, if someone in the future wanted to connect another database with a different format to our app, if we had used an adaptor pattern this would make it very easy.

Reply:
@joshuazeltser by different format, do you mean alternate database technologies such as (MySQL, PostgreSQL, etc.) or different versions of MySQL?

Reply:
It could be any of those. So it makes sense to use an adaptor pattern here so that someone could add in data from anywhere else